pkg/clusteragent/autoscaling/workload/config_retriever_settings_test.go

```diff
@@ -284,6 +294,183 @@ func TestConfigRetriverAutoscalingSettingsLeader(t *testing.T) {
+			{
+				Namespace: "ns",
+				Name:      "name1",
+				// Old .Spec field
+				Spec: objectv1alpha1Spec,
+			},
+			{
+				Namespace: "ns",
+				Name:      "name2",
+				Specs: &model.AutoscalingSpecs{
+					V1Alpha1: objectv1alpha1Spec,
+				},
+			},
+		},
+	}
+
+	// New Autoscaling settings received
+	stateCallbackCalled := 0
+	mockRCClient.triggerUpdate(
+		data.ProductContainerAutoscalingSettings,
+		map[string]state.RawConfig{
+			"foo1": buildAutoscalingSettingsRawConfig(t, 1, settingsList1),
+		},
+		func(_ string, applyState state.ApplyStatus) {
+			stateCallbackCalled++
+			assert.Equal(t, applyState, state.ApplyStatus{
+				State: state.ApplyStateAcknowledged,
+				Error: "",
+			})
+		},
+	)
+
+	assert.Equal(t, 1, stateCallbackCalled)
+	podAutoscalers := cr.store.GetAll()
+
+	// Expected v1alpha2 version output
+	objectv1alpha2Spec := &datadoghq.DatadogPodAutoscalerSpec{
+		Owner: datadoghqcommon.DatadogPodAutoscalerRemoteOwner,
+		TargetRef: autoscalingv2.CrossVersionObjectReference{
+			APIVersion: "v1",
+			Kind:       "Deployment",
+			Name:       "deploy1",
+		},
+		Objectives: []datadoghqcommon.DatadogPodAutoscalerObjective{
+			{
+				Type: datadoghqcommon.DatadogPodAutoscalerPodResourceObjectiveType,
+				PodResource: &datadoghqcommon.DatadogPodAutoscalerPodResourceObjective{
+					Name: corev1.ResourceCPU,
+					Value: datadoghqcommon.DatadogPodAutoscalerObjectiveValue{
+						Type:        datadoghqcommon.DatadogPodAutoscalerUtilizationObjectiveValueType,
+						Utilization: pointer.Ptr[int32](80),
+					},
+				},
+			},
+		},
+		ApplyPolicy: &datadoghq.DatadogPodAutoscalerApplyPolicy{
+			Mode: datadoghq.DatadogPodAutoscalerApplyModePreview,
+			ScaleUp: &datadoghqcommon.DatadogPodAutoscalerScalingPolicy{
+				Strategy: pointer.Ptr(datadoghqcommon.DatadogPodAutoscalerMinChangeStrategySelect),
+			},
+		},
+	}
+
+	// Copy with RemoteVersion set
+	expected := objectv1alpha2Spec.DeepCopy()
+	expected.RemoteVersion = pointer.Ptr[uint64](versionOffset + 1)
+	model.AssertPodAutoscalersEqual(t, []model.FakePodAutoscalerInternal{
+		{
+			Namespace:         "ns",
+			Name:              "name1",
+			Spec:              expected,
+			SettingsTimestamp: testTime,
+		},
+		{
+			Namespace:         "ns",
+			Name:              "name2",
+			Spec:              expected,
+			SettingsTimestamp: testTime,
+		},
+	}, podAutoscalers)
+
+	// Now the backend has upgraded to v2alpha1 keeping old v1alpha1 for old Cluster Agents
+	// v1alpha2 should be prioritzed, creating a small diff to check output
+	objectv1alpha2Spec.TargetRef.Name = "deploy2"
+	settingsList1 = model.AutoscalingSettingsList{
+		Settings: []model.AutoscalingSettings{
+			{
+				Namespace: "ns",
+				Name:      "name1",
+				Spec:      objectv1alpha1Spec,
+				Specs: &model.AutoscalingSpecs{
+					V1Alpha2: objectv1alpha2Spec,
+				},
+			},
+			{
+				Namespace: "ns",
+				Name:      "name2",
+				Spec:      objectv1alpha1Spec,
+				Specs: &model.AutoscalingSpecs{
```

Comment:
to test that `V1Alpha2` is prioritized over `V1Alpha1`, should `Specs` be modified to contain both?

Reply:
It's the old `.Spec` field, but I should also test the `v1alpha1` field` just in case